compose/service.py

```diff
@@ -678,10 +678,21 @@ def labels(self, one_off=False):
         ]
 
     def can_be_scaled(self):
-        for port in self.options.get('ports', []):
-            if ':' in str(port):
+        # TODO: this function should ideally move to docker.utils.ports
+        # get rid of it once its accepted into docker-py
+        def has_external_port(external_binding):
+            if external_binding is None:
                 return False
-        return True
+            if isinstance(external_binding, tuple) and \
+                    (len(external_binding) < 2 or external_binding[1] is None):
```

Comment:
I also doubt we need to check the length of the tuple given [docker-py's `split_port`](https://github.com/docker/docker-py/blob/master/docker/utils/ports/ports.py#L83) always returns a dual-length tuple (second element may be `None`, but it will still be there).